### PR TITLE
Numpy bool msgpack bugfix

### LINKED ISSUE #18390: to_msgpack fails for Dataframes with numpy bools
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import numpy as np
pd.DataFrame(pd.Series((np.bool_(1), 2, 3, 4, 5))).to_msgpack('test.msg')
```
#### Problem description

When trying to export Dataframes containing numpy bools with the DataFrame.to_msgpack() function one gets the exception:
```
  File "pandas/io/msgpack/_packer.pyx", line 230, in pandas.io.msgpack._packer.Packer.pack
  File "pandas/io/msgpack/_packer.pyx", line 232, in pandas.io.msgpack._packer.Packer.pack
  File "pandas/io/msgpack/_packer.pyx", line 191, in pandas.io.msgpack._packer.Packer._pack
  File "pandas/io/msgpack/_packer.pyx", line 220, in pandas.io.msgpack._packer.Packer._pack
  File "pandas/io/msgpack/_packer.pyx", line 191, in pandas.io.msgpack._packer.Packer._pack
  File "pandas/io/msgpack/_packer.pyx", line 220, in pandas.io.msgpack._packer.Packer._pack
  File "pandas/io/msgpack/_packer.pyx", line 227, in pandas.io.msgpack._packer.Packer._pack
TypeError: can't serialize True
```
I'm guessing this is because the isinstance check on [line 136 of pandas.io.msgpack._packer.pyx](https://github.com/pandas-dev/pandas/blob/1647a724ae5c9038cdd3cf4ebbf2273d1b6ab109/pandas/io/msgpack/_packer.pyx#L136) returns False for numpy.bool_ . 

#### Expected Output

I'd expect DataFrames with numpy bools in them to serialize just like ones with Python bools do.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.14.final.0
python-bits: 64
OS: Darwin
OS-release: 17.3.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: de_DE.UTF-8
LOCALE: None.None

pandas: 0.21.0
pytest: 3.1.2
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.25.2
numpy: 1.13.3
scipy: 0.19.1
pyarrow: 0.7.1
xarray: None
IPython: 5.3.0
sphinx: 1.6.2
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.2
feather: 0.4.0
matplotlib: 2.0.2
openpyxl: 2.4.7
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.8.0
bs4: 4.6.0
html5lib: 0.999
sqlalchemy: 1.1.11
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by PhyNerd**
This fixes issue #18390.
It allows for running DataFrame.to_msgpack() with dataframes containing fields of the numpy.bool_ datatype. 

- [x] closes #18390
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/io/msgpack/_packer.pyx [hunk 1]
```diff
@@ -133,7 +134,7 @@ cdef class Packer(object):
         while True:
             if o is None:
                 ret = msgpack_pack_nil(&self.pk)
-            elif isinstance(o, bool):
+            elif isinstance(o, (bool, bool_)):
```
Comment (jreback) [inline pandas/io/msgpack/_packer.pyx]: use ``np.bool_``

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/msgpack/test_pack.py [hunk 2]
```diff
@@ -25,6 +26,7 @@ def testPack(self):
             (), ((),), ((), None,),
             {None: 0},
             (1 << 23),
```
Comment (jreback) [inline pandas/tests/io/msgpack/test_pack.py]: can you add a round-trip test for the example in the issue itself.

Comment (PhyNerd) [reply pandas/tests/io/msgpack/test_pack.py]: This test should already test for the buggy behavior when it comes to `np.bool_`.

Should I revert the changes to this test and add a one checking round-trip behavior or should I make that test a additional one? 
Where would I best add this test? (There is no /test/io/formats/test_to_msgpack.py)

File: pandas/io/msgpack/_packer.pyx (hunk 1 above)
Comment (PhyNerd) [reply pandas/io/msgpack/_packer.pyx]: Will do.

**Comment by PhyNerd** [comment]
I applied the requested changes. 
I'm not sure why but one of the Travis CI builds timed-out.

**Comment by jreback** [comment]
can you add the tests from the original issue as well, e.g. the round-trip of the object array containing bools. rebase and ping on green.

**Comment by PhyNerd** [comment]
I added the requested check.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. just needs a rebase. ping when pushed.

**Comment by PhyNerd** [comment]
rebased.

**Comment by jreback** [comment]
thanks @PhyNerd

**Pull request merged**